### ISSUE TITLE
broken links from main listing when multiple packages with different required versions

### ISSUE THREAD

**Issue by joewiz** [eXist-db/public-xar-repo#7]
Clicking on either "EXPath Cryptographic Module Implementation" entry in http://exist-db.org/exist/apps/public-repo/index.html yields a server error from the resulting URL, http://exist-db.org/exist/apps/public-repo/packages/expath-crypto-exist-lib.html:

```
<exception>
  <path>/db/apps/public-repo/modules/view.xql</path>
  <message>exerr:ERROR XPTY0004: The actual cardinality for parameter 1 does not match the cardinality declared in the function's signature: app:package-to-list-item($app as element(), $show-details as xs:boolean) item()*. Expected cardinality: exactly one, got 2. [at line 35, column 35, source: /db/apps/public-repo/modules/app.xql]
In function:
	app:package-to-list-item(element(), xs:boolean) [35:9:/db/apps/public-repo/modules/app.xql]
	app:view-package(node(), map, xs:string?) [20:9:/db/apps/public-repo/modules/app.xql]
	templates:call-with-args(function, function*, element(), map) [208:13:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process-output(element(), map, item()*, element()) [205:9:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:call-by-introspection(element(), map, map, function) [187:28:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:call(item(), element(), map) [143:37:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [19:23:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	site:expand-links(node(), map, xs:string?) [20:9:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/siteutils.xql]
	templates:call-with-args(function, function*, element(), map) [208:13:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process-output(element(), map, item()*, element()) [205:9:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:call-by-introspection(element(), map, map, function) [187:28:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:call(item(), element(), map) [135:36:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [146:81:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [462:17:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:surround(node(), map, xs:string, xs:string?, xs:string?, xs:string?) [20:9:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:call-with-args(function, function*, element(), map) [208:13:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process-output(element(), map, item()*, element()) [205:9:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:call-by-introspection(element(), map, map, function) [187:28:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:call(item(), element(), map) [143:37:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [131:51:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:process(node()*, map) [88:9:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]
	templates:apply(node()+, function, map?, map?) [27:5:/home/existdb/exist-db.org/eXist-db.org/webapp/WEB-INF/data/expathrepo/shared-0.4.0/content/templates.xql]</message>
</exception>
```

I believe the fix involves passing additional parameters about required versions.

**Comment by joewiz** [comment]
Note that I've deployed the bug fix to exist-db.org, so now we can see the requests I cited in the original issue working. Starting from http://exist-db.org/exist/apps/public-repo/index.html, here are the URLs for the two versions of the EXPath Crypto module:

1. The "old" version (which lacks a <requires> element):

  http://exist-db.org/exist/apps/public-repo/packages/expath-crypto-exist-lib.html

2. The "new" version (which has a <requires semver-min="2.3.0"> element, which is translated into a new "eXist-db-min-version" URL parameter):

  http://exist-db.org/exist/apps/public-repo/packages/expath-crypto-exist-lib.html?eXist-db-min-version=2.3.0
